fix(make): runtime agnostic container detection

antoineco opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

Using `make` commands has been working poorly with non-Docker container runtimes (containerd, podman).

Checking the existence of the `/.dockerenv` file is only valid with the Docker runtime. A fairly reliable runtime agnostic alternative is to inspect the PID 1 cgroup. On most hosts, the PID 1 cgroup is the root cgroup (`/`).

Checking the first line of `/proc/1/sched` is another alternative, albeit less reliable.

nderjung reviewed (changes requested):
Really nice. Thanks @antoineco! I realise the verified git commit is more of a "DCO" than a sign-off, but to keep in convention with existing commits, could you please add your sign off (`git commit -s`) and force-push the new git commit.

Thanks!

antoineco committed:
fix(make): runtime agnostic container detection

Checking the existence of the /.dockerenv file is only valid with the
Docker runtime. A fairly reliable runtime agnostic alternative is to
inspect the PID 1 cgroup. On most hosts, the PID 1 cgroup is the root
cgroup (/).

Checking the first line of /proc/1/sched is another alternative, albeit
less reliable.

Signed-off-by: Antoine Cotten <antoine@unikraft.io>

antoineco commented:
@nderjung sure thing. Done 👍

nderjung commented:
🔥 🔥 🔥
